src/interface/web/app/components/loginPrompt/loginPrompt.tsx

```diff
@@ -1,47 +1,536 @@
+
+        const params = new URLSearchParams({
+            client_id: data.google.client_id,
+            redirect_uri: fullRedirectUri,
+            response_type: "code",
+            scope: "email profile openid",
+            state: window.location.pathname,
+            access_type: "offline",
+            prompt: "consent select_account",
+            include_granted_scopes: "true",
+        });
+
+        window.location.href = `https://accounts.google.com/o/oauth2/v2/auth?${params}`;
+    };
+
+    const handleGoogleScriptLoad = () => {
+        const google = (window as any).google;
+
+        if (!data?.google?.client_id || !data?.google?.redirect_uri) return;
+
+        // Initialize Google Sign In after script loads
+        google.accounts.id.initialize({
+            client_id: data?.google?.client_id,
+            callback: handleGoogleSignIn,
+            auto_select: false,
+            login_uri: data?.google?.redirect_uri,
+        });
+
+        // Render the button
+        google.accounts.id.renderButton(document.getElementById("g_id_signin")!, {
+            theme: "outline",
+            size: "large",
+            width: "100%",
+        });
+    };
+
+    function handleMagicLinkSignIn() {
+        fetch("/auth/magic", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({ email: email }),
+        })
+            .then((res) => {
+                if (res.ok) {
+                    setCheckEmail(true);
+                    if (checkEmail) {
+                        setRecheckEmail(true);
+                    }
+                    return res.json();
+                } else {
+                    throw new Error("Failed to send magic link");
+                }
+            })
+            .then((data) => {
+                console.log(data);
+            })
+            .catch((err) => {
+                console.error(err);
+            });
+    }
+
+    if (props.isMobileWidth) {
+        return (
+            <Drawer open={true} onOpenChange={props.onOpenChange}>
+                <DrawerContent className={`flex flex-col gap-4 w-full mb-4`}>
+                    <div>
+                        {useEmailSignIn && (
+                            <EmailSignInContext
+                                email={email}
+                                setEmail={setEmail}
+                                checkEmail={checkEmail}
+                                setCheckEmail={setCheckEmail}
+                                setUseEmailSignIn={setUseEmailSignIn}
+                                recheckEmail={recheckEmail}
+                                setRecheckEmail={setRecheckEmail}
+                                handleMagicLinkSignIn={handleMagicLinkSignIn}
+                            />
+                        )}
+                        {!useEmailSignIn && (
```

Comment:
Can this be simplified like so?

```suggestion
                        || (
```